pyrtl/memory.py

```diff
@@ -69,6 +70,10 @@ def sign_extended(self, bitwidth):
     def zero_extended(self, bitwidth):
         return as_wires(self).zero_extended(bitwidth)
 
+    @property
+    def name(self):
+        raise PyrtlError("MemIndexed is a temporary object and therefore doesn't have a name")
+
```

Comment:
Umm, read the text at the end again. the memindexed should make accesses and changes to a _MemIndexed.name affect the underlying wire, and make the wire if needed.